src-tauri/src/pipeline.rs

```diff
@@ -166,5 +178,42 @@ pub fn pipeline(
         vignetting_correction_cal,
     );
 
-    return vignetting_effect_correction_result;
+    if vignetting_effect_correction_result.is_err() {
+        return vignetting_effect_correction_result;
+    }
+
+    // Apply the neutral density filter.
+    let neutral_density_result = neutral_density(
+        &config_settings,
+        format!("{}output6.hdr", config_settings.temp_path),
+        format!("{}output7.hdr", config_settings.temp_path),
+        neutral_density_cal,
+    );
+
+    if neutral_density_result.is_err() {
+        return neutral_density_result
+    }
+
+    // Correct for photometric adjustments
+    let photometric_adjustment_result = photometric_adjustment(
+        &config_settings,
+        format!("{}output7.hdr", config_settings.temp_path),
+        format!("{}output8.hdr", config_settings.temp_path),
+        photometric_adjustment_cal,
+    );
+
+    if photometric_adjustment_result.is_err() {
+        return photometric_adjustment_result
+    }
+
+    // Edit the header
+    let header_editing_result = header_editing(
+        &config_settings,
+        format!("{}output8.hdr", config_settings.temp_path),
+        format!("{}output9.hdr", config_settings.output_path),
```

Comment:
This causes an error for me... if I change line 213 to `format!("{}output9.hdr", config_settings.temp_path),` instead, it works since we're only using the temp path at the moment